test/hotspot/jtreg/vmTestbase/nsk/share/jvmti/hotswap/HotSwap.cpp

```diff
@@ -96,26 +98,36 @@ ClassFileLoadHook(jvmtiEnv *jvmti_env, JNIEnv *jni_env,
         jint *new_class_data_len, unsigned char** new_class_data) {
     jint name_len;
 
-    if (name != NULL && classCount < max_classes &&
+    if (name != NULL &&
             class_being_redefined == NULL &&
             (strcmp(name, PROFILE_CLASS_NAME) != 0) &&
             (strncmp(name, package_name, package_name_length) == 0)) {
-        NSK_DISPLAY1("ClassFileLoadHook: %s\n", name);
-        name_len = (jint) strlen(name) + 1;
-        if (!NSK_JVMTI_VERIFY(jvmti_env->Allocate(name_len, (unsigned char**) &names[classCount]))) {
+        if (!NSK_JVMTI_VERIFY(jvmti_env->RawMonitorEnter(classLoadLock))) {
             nsk_jvmti_setFailStatus();
             return;
         }
-        memcpy(names[classCount], name, name_len);
-        if (!NSK_JVMTI_VERIFY(jvmti_env->Allocate(class_data_len, (unsigned char**)
-                &old_class_def[classCount].class_bytes))) {
+        // use while instead of if to exit the block on error
+        while (classCount < max_classes) {
```

Comment:
Could you please explain why this while is executed only once. It is not obvious.

Reply:
I tried to explain it in the comment :)
it should be "if (classCount < max_classes)",
but inside the if we have return statements and we need to do cleanup (exit monitor).
So I used well-known pattern with 1-pass cycle (note that "return"s are replaced with "break"s)

Reply:
Offline discussion shown that I forgot to add break at the end of the cycle. Will fix